### DISCUSSION TITLE
Passing stepper content templates dynamically

### DISCUSSION THREAD

**Discussion by Bond-Addict** [epicmaxco/vuestic-ui#3639] [Q&A] [answered]
Is it possible to pass templates dynamically to the stepper? I've created a custom implementation of the VaStepper and everything works except for the stepper content. I would think I could pass them to a slot.

Click on Stepper => Look at "Custom Stepper" section.
https://codesandbox.io/p/sandbox/nice-darwin-iqjx2m

**Comment by m0ksem** [comment, marked as answer]
Hi. Sure, you can use [Dynamic slots](https://vuejs.org/guide/components/slots.html#dynamic-slot-names) and proxy all incomming slots to `va-stepper`.

```vue
<va-stepper>
  <template
    v-for="name in $slots"
    :key="name"
    v-slot:[name]="slotScope"
  >
    <slot
      :name="name"
      v-bind="{ ...slotScope,  }"
    />
  </template>
</va-stepper>
```

You also can filter slots which not starts with `step` if you need.
